From: Alexandru Avadanii <Alexandru.Avadanii@enea.com>
Date: Mon, 21 Aug 2017 22:26:09 +0200
Subject: [PATCH] reclass: baremetal: openstack: Fix eth assignment

Baremetal support introduced a couple of VCP VMs, which have 2
network interfaces:
- primary (ens3 inside x86 VM) - connected to "br-mgmt" bridge on
  each kvm node, serves for MaaS DHCP / connection to salt master;
- secondary (ens4 inside x86 VM) - connected to "br-ctl" bridge on
  each kvm node, serves for Openstack Management network;

However, the reclass model was configured to use a single IP address
on the primary interface, breaking the connnection to salt master,
while also not connecting the Openstack Management network properly.

Fix this by configuring the primary interface for DHCP, while the
secondary gets a static IP in Openstack Management network.

Change-Id: I9f1d6f080e882bfaae7b5f209bc3c5536826ba06
Signed-off-by: Alexandru Avadanii <Alexandru.Avadanii@enea.com>
---
 .../classes/cluster/baremetal-mcp-ocata-ovs-ha/openstack/control.yml | 5 +++--
 .../cluster/baremetal-mcp-ocata-ovs-ha/openstack/dashboard.yml       | 3 ++-
 .../cluster/baremetal-mcp-ocata-ovs-ha/openstack/database.yml        | 5 +++--
 .../cluster/baremetal-mcp-ocata-ovs-ha/openstack/message_queue.yml   | 5 +++--
 .../classes/cluster/baremetal-mcp-ocata-ovs-ha/openstack/proxy.yml   | 5 +++--
 .../cluster/baremetal-mcp-ocata-ovs-ha/openstack/telemetry.yml       | 5 +++--
 6 files changed, 17 insertions(+), 11 deletions(-)

diff --git a/mcp/reclass/classes/cluster/baremetal-mcp-ocata-ovs-ha/openstack/control.yml b/mcp/reclass/classes/cluster/baremetal-mcp-ocata-ovs-ha/openstack/control.yml
index 995c50c..20b470d 100644
--- a/mcp/reclass/classes/cluster/baremetal-mcp-ocata-ovs-ha/openstack/control.yml
+++ b/mcp/reclass/classes/cluster/baremetal-mcp-ocata-ovs-ha/openstack/control.yml
@@ -23,7 +23,7 @@ classes:
 - cluster.baremetal-mcp-ocata-ovs-ha.infra
 parameters:
   _param:
-    keepalived_vip_interface: ens3
+    keepalived_vip_interface: ens4
     keepalived_vip_virtual_router_id: 50
     cluster_vip_address: ${_param:openstack_control_address}
     cluster_local_address: ${_param:single_address}
@@ -37,7 +37,8 @@ parameters:
   linux:
     network:
       interface:
-        ens3: ${_param:linux_single_interface}
+        ens3: ${_param:linux_dhcp_interface}
+        ens4: ${_param:linux_single_interface}
   bind:
     server:
       control:
diff --git a/mcp/reclass/classes/cluster/baremetal-mcp-ocata-ovs-ha/openstack/dashboard.yml b/mcp/reclass/classes/cluster/baremetal-mcp-ocata-ovs-ha/openstack/dashboard.yml
index b7ed814..522719a 100644
--- a/mcp/reclass/classes/cluster/baremetal-mcp-ocata-ovs-ha/openstack/dashboard.yml
+++ b/mcp/reclass/classes/cluster/baremetal-mcp-ocata-ovs-ha/openstack/dashboard.yml
@@ -7,4 +7,5 @@ parameters:
   linux:
     network:
       interface:
-        ens3: ${_param:linux_single_interface}
+        ens3: ${_param:linux_dhcp_interface}
+        ens4: ${_param:linux_single_interface}
diff --git a/mcp/reclass/classes/cluster/baremetal-mcp-ocata-ovs-ha/openstack/database.yml b/mcp/reclass/classes/cluster/baremetal-mcp-ocata-ovs-ha/openstack/database.yml
index c0e21aa..8aa8b70 100644
--- a/mcp/reclass/classes/cluster/baremetal-mcp-ocata-ovs-ha/openstack/database.yml
+++ b/mcp/reclass/classes/cluster/baremetal-mcp-ocata-ovs-ha/openstack/database.yml
@@ -16,7 +16,7 @@ classes:
 - cluster.baremetal-mcp-ocata-ovs-ha
 parameters:
   _param:
-    keepalived_vip_interface: ens3
+    keepalived_vip_interface: ens4
     keepalived_vip_virtual_router_id: 80
     galera_server_cluster_name: openstack_cluster
     cluster_vip_address: ${_param:openstack_database_address}
@@ -30,4 +30,5 @@ parameters:
   linux:
     network:
       interface:
-        ens3: ${_param:linux_single_interface}
+        ens3: ${_param:linux_dhcp_interface}
+        ens4: ${_param:linux_single_interface}
diff --git a/mcp/reclass/classes/cluster/baremetal-mcp-ocata-ovs-ha/openstack/message_queue.yml b/mcp/reclass/classes/cluster/baremetal-mcp-ocata-ovs-ha/openstack/message_queue.yml
index 3b79030..0840cc4 100644
--- a/mcp/reclass/classes/cluster/baremetal-mcp-ocata-ovs-ha/openstack/message_queue.yml
+++ b/mcp/reclass/classes/cluster/baremetal-mcp-ocata-ovs-ha/openstack/message_queue.yml
@@ -7,7 +7,7 @@ classes:
 - cluster.baremetal-mcp-ocata-ovs-ha
 parameters:
   _param:
-    keepalived_vip_interface: ens3
+    keepalived_vip_interface: ens4
     keepalived_vip_virtual_router_id: 90
     cluster_vip_address: ${_param:openstack_message_queue_address}
     cluster_local_address: ${_param:single_address}
@@ -20,4 +20,5 @@ parameters:
   linux:
     network:
       interface:
-        ens3: ${_param:linux_single_interface}
+        ens3: ${_param:linux_dhcp_interface}
+        ens4: ${_param:linux_single_interface}
diff --git a/mcp/reclass/classes/cluster/baremetal-mcp-ocata-ovs-ha/openstack/proxy.yml b/mcp/reclass/classes/cluster/baremetal-mcp-ocata-ovs-ha/openstack/proxy.yml
index 2695c96..86f331d 100644
--- a/mcp/reclass/classes/cluster/baremetal-mcp-ocata-ovs-ha/openstack/proxy.yml
+++ b/mcp/reclass/classes/cluster/baremetal-mcp-ocata-ovs-ha/openstack/proxy.yml
@@ -15,7 +15,7 @@ classes:
 # - cluster.baremetal-mcp-ocata-ovs-ha.stacklight.proxy
 parameters:
   _param:
-    keepalived_vip_interface: ens3
+    keepalived_vip_interface: ens4
     keepalived_vip_virtual_router_id: 240
     nginx_proxy_ssl:
       enabled: true
@@ -27,7 +27,8 @@ parameters:
   linux:
     network:
       interface:
-        ens3: ${_param:linux_single_interface}
+        ens3: ${_param:linux_dhcp_interface}
+        ens4: ${_param:linux_single_interface}
     system:
       package:
         libapache2-mod-wsgi:
diff --git a/mcp/reclass/classes/cluster/baremetal-mcp-ocata-ovs-ha/openstack/telemetry.yml b/mcp/reclass/classes/cluster/baremetal-mcp-ocata-ovs-ha/openstack/telemetry.yml
index ca655dd..b46319f 100644
--- a/mcp/reclass/classes/cluster/baremetal-mcp-ocata-ovs-ha/openstack/telemetry.yml
+++ b/mcp/reclass/classes/cluster/baremetal-mcp-ocata-ovs-ha/openstack/telemetry.yml
@@ -10,7 +10,7 @@ classes:
 - cluster.baremetal-mcp-ocata-ovs-ha.infra
 parameters:
   _param:
-    keepalived_vip_interface: ens3
+    keepalived_vip_interface: ens4
     keepalived_vip_virtual_router_id: 230
     cluster_vip_address: ${_param:openstack_telemetry_address}
     cluster_local_address: ${_param:single_address}
@@ -23,4 +23,5 @@ parameters:
   linux:
     network:
       interface:
-        ens3: ${_param:linux_single_interface}
+        ens3: ${_param:linux_dhcp_interface}
+        ens4: ${_param:linux_single_interface}
